pyro/ops/einsum/torch_log.py

```diff
@@ -29,12 +29,12 @@ def einsum(equation, *operands):
     shifts = []
     exp_operands = []
     for dims, operand in zip(inputs, operands):
-        shift = operand
+        shift = operand.detach()
         for i, dim in enumerate(dims):
             if dim not in output:
                 shift = shift.max(i, keepdim=True)[0]
         # avoid nan due to -inf - -inf
-        shift = shift.clamp(min=torch.finfo(shift.dtype).min)
+        shift.clamp_(min=torch.finfo(shift.dtype).min)
```

Comment:
Looks like this in-place op is triggering a bunch of CI failures:
```
RuntimeError: unsupported operation: more than one element of the written-to tensor refers to a single memory location. Please clone() the tensor before performing the operation.
```

Reply:
Thanks, reverted!